### PR TITLE
make: use code style CFLAGS from Makefile.cflags

### PR DESCRIPTION
**Pull request by gebart**
Since these flags are defined unconditionally in Makefile.cflags there's no reason to add them a second time in the board specific Makefiles.

### CODE REVIEW

File: boards/arduino-mega2560/Makefile.include [hunk 1]
```diff
@@ -45,9 +45,9 @@ endif
 
 # define build specific options
 export CPU_USAGE = -mmcu=atmega2560
-export CFLAGS += -ggdb -g3 -std=gnu99 -Os -Wall -Wstrict-prototypes $(CPU_USAGE)
+export CFLAGS += -ggdb -g3 -Os $(CPU_USAGE)
```
Comment (kaspar030) [inline boards/arduino-mega2560/Makefile.include]: Could you use CFLAGS_OPT for -Os?

Comment (kaspar030) [inline boards/arduino-mega2560/Makefile.include]: (Sorry, might be too unrelated)

Comment (gebart) [inline boards/arduino-mega2560/Makefile.include]: I'll update avr in a separate commit, but this PR.

**Comment by gebart** [comment]
@kaspar030 addressed comments and homogenized all CFLAGS settings for some platforms (cortexm_common, msp430_common, arduino-mega2560)

**Comment by gebart** [comment]
(I don't really want to spend any time on the arm7 platforms because in my opinion they're old as crap and don't have much of a use case in today's IoT world)

**Comment by OlegHahm** [comment]
Is there a reason to keep the variable at all?

**Comment by gebart** [comment]
@OlegHahm I guess we can drop the CFLAGS_STYLE variable.

**Comment by OlegHahm** [comment]
:+1: for dropping

**Comment by gebart** [comment]
Dropped the CFLAGS_STYLE variable and updated msba2-common to follow other platforms more closely with regards to CFLAGS and the Makefile layout. (call it OCD or whatever, but it felt wrong leaving them out when updating the Makefiles..)

Added automatic selection of newlib-nano if available for msba2-common as well (all cortex platforms already use it if it is available)

**Comment by gebart** [comment]
@OlegHahm was that an ACK?

**Comment by OlegHahm** [comment]
Not yet.

**Comment by OlegHahm** [comment]
Can someone test on the arduino-mega?

**Comment by OlegHahm** [comment]
ACK for the rest.

**Comment by OlegHahm** [comment]
@haukepetersen, don't you have one on your desk? Can you just try to flash default example and check if it still works? Then we can merge this.

**Comment by kYc0o** [comment]
I just get one, I´ll test it asap

**Comment by gebart** [comment]
rebased on top of latest master (after merging #4778)

**Comment by OlegHahm** [comment]
@kYc0o, awesome, thanks!

**Comment by kYc0o** [comment]
compiled and flashed default example correctly on arduino-mega2560

**Comment by OlegHahm** [comment]
In this case, merge at will as soon as Travis agrees.

**Comment by kYc0o** [comment]
Travis is not happy :/
```
periph_rtt:airfy-beacon:
Building application "periph_rtt" for "airfy-beacon" with MCU "nrf51".
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/bin/airfy-beacon/periph_rtt.a(main.o): In function `cb':
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:38: undefined reference to `rtt_get_counter'
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:40: undefined reference to `rtt_set_alarm'
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/bin/airfy-beacon/periph_rtt.a(main.o): In function `main':
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:51: undefined reference to `rtt_init'
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:54: undefined reference to `rtt_set_alarm'
collect2: error: ld returned 1 exit status
make: *** [all] Error 1
```

**Comment by OlegHahm** [comment]
Let's see if we can make it happy by restarting.

**Comment by gebart** [comment]
Travis is happy -> go

**Pull request merged**